### PR TITLE
chore(ci): update ubuntu machines to 20.04 COMPASS-5637

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch updates all Ubuntu machines we are using in evergreen CI to 20.04 distro.

This allows us to enable e2e tests for mongodb 5 and 6 on Ubuntu, but also required me to disable 4.0 and 4.4 enterprise e2e tests at the same time. Seems like mongodb toolchain might be missing a dependency required for mongodb to work on this version of Ubuntu correctly, it [fails](https://spruce.mongodb.com/task/10gen_compass_main_ubuntu_test_packaged_app_40x_enterprise_patch_c4b6f0d859d47fb2b0f163fb5b051bdd974d73f0_63beda740305b96865a1d8e6_23_01_11_15_49_26/logs?execution=0) with `error while loading shared libraries: libnetsnmpmibs.so.30: cannot open shared object file: No such file or directory`. This seems like an okay compromise to me for now and we can figure out how to deal with this later, but please tell me if you feel otherwise about it.

Partially for the same reason I changed default mongodb version for unit / functional tests to 6.x too, IIRC the idea there was always to run these types of tests against latest-ish, we just never updated the version.

Here's an [evergreen patch](https://spruce.mongodb.com/version/63bfdd0357e85a1deec28ccb/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) I manually spawned to test that everything works before opening a PR

### CODE REVIEW

**Comment by addaleax** [comment]
> This seems like an okay compromise to me for now and we can figure out how to deal with this later, but please tell me if you feel otherwise about it.

Just wondering, if it’s easy enough, could we also keep the older enterprise versions by using `run_on:` per-task? If not, I think that’s totally fine, yes.

**Comment by gribnoysup** [comment]
Yeah, we can do that too (I just wasn't feeling particularly strong about doing it), and probably can make it at least 18, seems like mongosh tests are using this version for 4.x. Will update, thanks for the feedback!

**Comment by gribnoysup** [comment]
Updated, seems to be [running fine](https://spruce.mongodb.com/version/63c000685623436e925e2a6c/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) on ubuntu 18 (still running, but didn't crash immediately as on ubuntu 20)

**Review by addaleax — APPROVED** [review]

**Comment by gribnoysup** [comment]
Not sure yet what exactly is up, but seems like kerberos is now not bundling correctly, the package is failing with the following error on import:

```
/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /data/mci/43d292001ad02422635d69aef105b5e9/src/packages/compass/dist/MongoDB Compass Dev-linux-x64/resources/app.asar.unpacked/node_modules/kerberos/build/Release/kerberos.node
```

So possibly we can't really stop using Ubuntu 16. Will look into this more later

**Comment by addaleax** [comment]
Oh, yeah, sorry I missed that. Yes, compiling/re-building addons needs to happen on the platform with the lowest glibc version, always, i.e. those tasks actually need to remain on Ubuntu 16.04, otherwise that would break users on that platform (and equivalent ones).

**Comment by gribnoysup** [comment]
Aha, thank you for taking a closer look and explanation! I'll update package tasks to work from older Ubuntu then 👍

File: packages/compass-e2e-tests/tests/logging.test.ts [hunk 1]
```diff
@@ -366,7 +366,12 @@ describe('Logging and Telemetry integration', function () {
 
       it('does not contain warnings about missing optional dependencies', function () {
         const ids = logs.map(({ id }) => id);
-        expect(ids).not.to.contain(1_000_000_041);
+        expect(ids).not.to.contain(
+          1_000_000_041,
+          `Expected log to not contain warnings about missing dependencies, but got ${JSON.stringify(
```
Comment (gribnoysup) [inline packages/compass-e2e-tests/tests/logging.test.ts:371]: This is not really related to the changes, but made it much easier to debug what's going on, doesn't hurt keeping this I think
